content/en/tracing/trace_collection/_index.md

```diff
@@ -27,47 +26,98 @@ To configure your application to send traces to Datadog:
+
+   ```
+   DD_TRACE_AGENT_URL=http://custom-hostname:1234
+   DD_TRACE_AGENT_URL=unix:///var/run/datadog/apm.socket
+
+   ```
+
+   Similarly, the trace client attempts to send stats to the `/var/run/datadog/dsd.socket` Unix domain socket. If the socket does not exist, then stats are sent to `http://localhost:8125`.
+
+{{< site-region region="us3,us5,eu,gov,ap1" >}}
 
-<br/>
+4. Set `DD_SITE` in the Datadog Agent to {{< region-param key="dd_site" code="true" >}} to ensure the Agent sends data to the right Datadog location.
 
+{{< /site-region >}}
 
+[1]: /agent/guide/agent-configuration-files/#agent-main-configuration-file
+{{% /tab %}}
+{{% tab "AWS Lambda" %}}
+
+To set up Datadog APM in AWS Lambda, see the [Tracing Serverless Functions][1] documentation.
+
+[1]: /tracing/serverless_functions/
+{{% /tab %}}
+{{% tab "Other Environments" %}}
+
+Tracing is available for several other environments, such as  [Heroku][1], [Cloud Foundry][2], [AWS Elastic Beanstalk][3], and [Azure App Service][4].
+
+For other environments, see the [Integrations][5] documentation for that environment and [contact support][6] if you are encountering any setup issues.
+
+[1]: /agent/basic_agent_usage/heroku/#installation
+[2]: /integrations/cloud_foundry/#trace-collection
+[3]: /integrations/amazon_elasticbeanstalk/
+[4]: /infrastructure/serverless/azure_app_services/#overview
+[5]: /integrations/
+[6]: /help/
+{{% /tab %}}
+{{< /tabs >}}
+
+For more language-specific details, see the [Tracing Library][4] pages.
 
 ## Step 2 - Instrument your application 
 
 When you add the Datadog tracing library to your code, it instruments the service and sends traces to the Datadog Agent. The Agent then sends the traces to the Datadog backend to be displayed in the UI.
 
 {{< img src="tracing/visualization/troubleshooting_pipeline.png" alt="The APM pipeline">}}
 
-Depending on the programming language and infrastructure you use, you have the following options to instrument your application.
+Depending on the programming language and infrastructure you use, you have the following options to instrument your application:
 
-### Option 1 - Add the library directly to your application code
+- [Inject the library locally without modifying your application code](#option-2---inject-the-library-locally-without-modifying-code)
+- [Add the library directly to your application code](#option-1---add-the-library-directly-to-your-application-code)
+- [Inject the library from the Datadog UI](#option-3---inject-the-library-from-the-datadog-ui)
 
-For setup instructions, select your language:
+### Option 1 - Inject the library locally without modifying your application code
 
-{{< partial name="apm/apm-languages.html" >}}
+For Kubernetes, hosts, and containers, you can inject the tracing library locally without modifying your application code. For more information and instructions, read [Injecting Libraries Locally][2].
```

Comment:
I think we need to better explain what "injecting the tracing library means", something like "For Kubernetes, hosts, and containers you can auto-instrument your application by injecting the tracing library into your process."